### ISSUE TITLE
FileUploader - screen reader and keyboard accessibility vioations

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#3256]
From https://github.com/carbon-design-system/carbon/pull/3175#issuecomment-507386679

_Testing drag and drop variant in React, but issues apply across all variants of file uploader across packages_

_Tested on Windows 10 and macOS latest in Chrome and Firefox latest using VoiceOver and NVDA_

![Jul-01-2019 11-30-50](https://user-images.githubusercontent.com/40970507/60452342-ef8f1280-9bf3-11e9-9789-2186d58ba6fb.gif)

Here you see a drag and drop file uploader with two files uploaded. One failed upload and one successful. The first tab stop in this component is the link to upload a file. The user misses all the information about file size limits, file type etc. This should be communicated via an `aria-label` _or maybe consider making the larger component (`div` or whatever) a tab stop that communicates "hey, this is a file uploader, files uploaded should be less than 500kb, there are currently no uploaded files"._

When the user uploads a file it's success or failure isn't communicated. This information is obviously pretty important to getting the job done. You could use an `aria-live` region to communicate this error -- here's a good resource
https://cantina.co/form-errors-screen-readers-can-access/

The last big thing is the list of uploaded files. Just brain-deving but my thinking is the outer most element for that section should be a `ul` with a label that communicates what it's a list of (uploaded files). Each uploaded file then becomes an `li` that the user should be able to tab through for more detailed information (name of file, status of upload) or to delete the uploaded file. 

Aside from that there is the matter of having the SVG X file delete button described as an image for  a resource on how to handle that check out this article:
https://www.scottohara.me/blog/2019/05/22/contextual-images-svgs-and-a11y.html

For a great resource on the general expectation of a file uploader as far as accessibility is concerned take a look at some of the interactions described here
https://design-system.service.gov.uk/components/file-upload/

---
### TLDR

- [ ] initial tab stop doesn't describe requirements and is too terse
- [ ] success and failure states aren't communicated to screen reader user
- [ ] uploaded files should be a `ul` with each file an `li` including proper aria-labels
- [ ] file delete button should be described as a "button" with it's functionality described

**Comment by dakahn** [comment]
@emyarod 👍🏽

**Comment by abbeyhrt** [comment]
Since #3438 will be closed, just wanted to toss this note from @snidersd here so it doesn't get lost "Pressing the button clears the file, but the user is unaware that is happening." Nothing is announced to the screenreader when the file is successfully deleted.

**Comment by emyarod** [comment]
this issue was opened against the draft/in progress PR. do all of the original points apply to the released component?

**Comment by joshblack** [comment]
Revisited this today with DA who confirmed that this was not necessarily a bug, but rather a preferred way of working with this control. Moving over to our proposals pipeline :+1:

**Comment by aledavila** [comment]
@dakahn please research this

**Comment by designertyler** [comment]
Closing. @dakahn  Will open a new one if this feedback still applies to the latest file uploader.
